http-api/src/main/java/io/avaje/http/api/context/ThreadLocalResolver.java

```diff
@@ -0,0 +1,54 @@
+package io.avaje.http.api.context;
+
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import java.util.function.Supplier;
+
+public final class ThreadLocalResolver implements RequestContextResolver {
```

Comment:
This is a little nit-picky but I feel like the impl class name should be `ThreadLocalRequestContextResolver`, if you just saw `new ThreadLocalResolver()` used in some code to set the global resolver, then it’s not really obvious that the class is a RequestContextResolver implementation.